plugin.rb

```diff
@@ -533,6 +534,9 @@ def self.onebox_template
     put '/chat_channels/:chat_channel_id/change_status' => 'chat_channels#change_status'
     delete '/chat_channels/:chat_channel_id' => 'chat_channels#destroy'
 
+    # hints controller. Only used by staff members, we don't want to leak category permissions.
+    get '/hints/:category_id' => 'hints#access_by_category', format: :json, constraints: StaffConstraint.new
```

Comment:
I have been thinking quite a long time about this and I think the way forward is:

(you can use chat_channels for now, I want to go dash for everything in the future)

- POST/PUT /chat/api/draft-chat-channels/
- which return a draft id, a chatable type, chatable id, name, description, and hints in a meta key
- when you choose a category for the first time it creates the draft-channel, if you change the category (and have a draft id) you PUT it to get the new hints

In the future everything will be in the same screen (creating DM and creating public channels) and we will have these URLs: /chat/draft-channels/new and /chat/draft-channels/:id/edit

Let me know what you think of this plan

Reply:
I understand the benefit of having a draft for a topic or post, where you don't want to lose what you wrote or pause and later come back at it, but I don't see much value in having the same for channels. Maybe I'm missing something, but creating a channel should be quick and easy, even if you lost the form values and have to start over.